dask_kubernetes/operator/controller/controller.py

```diff
@@ -461,7 +461,14 @@ async def get_desired_workers(scheduler_service_name, namespace, logger):
 worker_group_scale_locks = defaultdict(lambda: asyncio.Lock())
 
 
+def resource_is_deleted(event, **_):
+    return event["type"] == "DELETED"
+
+
 @kopf.on.field("daskworkergroup.kubernetes.dask.org", field="spec.worker.replicas")
+@kopf.on.event(
+    kind="pod", when=resource_is_deleted, labels={"dask.org/component": "worker"}
+)
```

Comment:
I think with this decorator the `spec` that will be passed to `daskworkergroup_replica_update` will be the spec for the Pod, not the DaskWorkerGroup.

I think we probably need to put this on a separate function that gets the spec for the worker group and then calls `daskworkergroup_replica_update`.